modules/exploits/linux/http/cacti_unauthenticated_cmd_injection.rb

```diff
@@ -0,0 +1,300 @@
+      update_info(
+        info,
+        'Name' => 'Cacti 1.2.22 unauthenticated command injection',
+        'Description' => %q{
+          This module exploits an unauthenticated command injection
+          vulnerability in Cacti through 1.2.22 (CVE-2022-46169) in
+          order to achieve unauthenticated remote code execution as the
+          www-data user.
+
+          The module first attempts to obtain the Cacti version to see
+          if the target is affected. If LOCAL_DATA_ID and/or HOST_ID
+          are not set, the module will try to bruteforce the missing
+          value(s). If a valid combination is found, the module will
+          use these to attempt exploitation. If LOCAL_DATA_ID and/or
+          HOST_ID are both set, the module will immediately attempt
+          exploitation.
+
+          During exploitation, the module sends a GET request to
+          /remote_agent.php with the action parameter set to polldata
+          and the X-Forwarded-For header set to the provided value for
+          X_FORWARDED_FOR_IP (by default 127.0.0.1). In addition, the
+          poller_id parameter is set to the payload and the host_id
+          and local_data_id parameters are set to the bruteforced or
+          provided values. If X_FORWARDED_FOR_IP is set to an address
+          that is resolvable to a hostname in the poller table, and the
+          local_data_id and host_id values are vulnerable, the payload
+          set for poller_id will be executed by the target.
+
+          This module has been successfully tested against Cacti
+          version 1.2.22 running on Ubuntu 21.10 (vulhub docker image)
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Stefan Schiller', # discovery (independent of Steven Seeley)
+          'Steven Seeley', # (mr_me) @steventseeley - discovery (independent of Stefan Schiller)
+          'Owen Gong', # @phithon_xg - vulhub PoC
+          'Erik Wynter' # @wyntererik - Metasploit
+        ],
+        'References' => [
+          ['CVE', '2022-46169'],
+          ['URL', 'https://github.com/Cacti/cacti/security/advisories/GHSA-6p93-p743-35gf'], # disclosure and technical details
+          ['URL', 'https://github.com/vulhub/vulhub/tree/master/cacti/CVE-2022-46169'], # vulhub vulnerable docker image and PoC
+          ['URL', 'https://www.sonarsource.com/blog/cacti-unauthenticated-remote-code-execution'] # analysis by Stefan Schiller
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 8080
+        },
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Automatic (Unix In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' },
+              'Type' => :unix_memory
+            }
+          ],
+          [
+            'Automatic (Linux Dropper)',
```

Comment:
Sorry that I missed this in the earlier review. The `CmdStagerFlavor` option should be populated with the correct flavors that work and match the platform.

Reply:
Went ahead and tested / added the command stager flavors in a commit. Will land this in a bit!